crates/fuel-core/src/service/adapters/txpool.rs

```diff
@@ -141,9 +141,10 @@ impl fuel_core_txpool::ports::TxPoolDb for Database {
     }
 }
 
+#[async_trait::async_trait]
 impl GasPriceProvider for StaticGasPrice {
-    fn gas_price(&self, _block_height: BlockHeight) -> Option<u64> {
-        Some(self.gas_price)
+    async fn gas_price(&self, _block_bytes: u64) -> TxPoolResult<u64> {
```

Comment:
Hmm, for TxPool I think it is okay to use the last gas price(we don't need to predict it).